main.js

```diff
@@ -0,0 +1,286 @@
+
+    let playerOneSelections = [];
+    let playerTwoSelections = [];
+
+    function resetPlayers() {
+        this.playerOneSelections = [];
+        this.playerTwoSelections = [];
+    };
+
+    return {
+        playerOneSelections,
+        playerTwoSelections,
+        resetPlayers,
+    };
+})();
+
+
+
+/////////////////////////////////////////////
+
+
+
+
+const Player = (name, mark) => {
+
+    
+    return {
+        name,
+        mark,
+    };
+};
+
+const player1 = Player('Player 1', 'X');
+const player2 = Player('Player 2', 'O');
+
+
+
+//////////////////////////////////////
+
+
+
+const game = (() => {
+    
+    const playerTurn = document.getElementById('player-turn');
+    playerTurn.style.display = "none";
+    let x_turn = true;
+    let roundCount = 0;
+    let player1name;
+    let player2name;
+    const winningCombinations = [
+        [0, 1, 2],
+        [3, 4, 5],
+        [6, 7, 8],
+        [0, 3, 6],
+        [1, 4, 7],
+        [2, 5, 8],
+        [0, 4, 8],
+        [2, 4, 6],
+    ];
+
+    const cellElements = document.querySelectorAll('[data-cell]');
+    let cellArray = Array.from(cellElements);
+
+
+
+    // create module to control display
+    const displayController = (() => {
+
+        // select for "home screen" buttons
+        const restartBtn = document.getElementById('restart-btn');
+        restartBtn.addEventListener('click', resetDisplay);
+        restartBtn.style.display = 'none';
+        restartBtn.addEventListener('click', resetDisplay);
+            function resetDisplay () {
+                playerTurn.innerText = '';
+                playerTurn.style.display = 'none';
+                playerOneNameInput.style.display = 'inline';
+                playerTwoNameInput.style.display = 'inline';
+                startGame.style.display = 'inline';
+                startBtn.style.display = 'inline';
+                restartBtn.style.display = 'none';
+                cellElements.forEach(cell => {
+                    cell.innerText = '';
+                    cell.setAttribute('id', '');
+                    cell.setAttribute('class', 'cell');
+                });
+            }
+
+        const startGame = document.forms['start-game-form'];
+        startGame.addEventListener('submit', startFunction);
+        const playerOneNameInput = startGame.querySelector('[data-player-1-input]')
+        const playerTwoNameInput = startGame.querySelector('[data-player-2-input]')
+
+
+        const startBtn = document.getElementById('start-game-btn');
+    
+        // create function to hide home screen and start game
+        function startFunction(e) {
+            e.preventDefault();
+            activateCells();
+            let playerOneNameInputVal = startGame.querySelector('[data-player-1-input]').value;
+            let playerTwoNameInputVal = startGame.querySelector('[data-player-2-input]').value;
+            player1name = playerOneNameInputVal;
+            player2name = playerTwoNameInputVal;
```

Comment:
to expand on this defaulting back to X's turn or O's turn you would need to wrap these in a conditional to see if there is any value before assigning that value to the variable.